internal/provider/newrelic/provider_integration_test.go

```diff
@@ -168,12 +177,155 @@ func Test_Getting_external_metric_through_proxy(t *testing.T) {
+//
+//nolint:funlen,tparallel,paralleltest // no parallel since port 3000 is used by all test.
+func Test_does_not_hide_backend_errors(t *testing.T) {
+	t.Parallel()
 
-	testEnv := testutil.TestEnv{}
+	ctx := testutil.ContextWithDeadline(t)
+
+	testEnv := testutil.TestEnv{
+		StartKubernetes: true,
+		Region:          "Local",
+	}
 	testEnv.Generate(t)
 
+	url := fmt.Sprintf("%s/apis/external.metrics.k8s.io/v1beta1/namespaces/*/test_metric", testEnv.BaseURL)
+
+	p := newrelicProviderWithMetric(t, newrelic.Metric{
+		Query: testIntegrationQuery,
+	}, &testEnv)
+
+	runAdapter(t, &testEnv, p)
+
+	t.Run("when_newrelic_backend_response_error_is", func(t *testing.T) {
+		cases := map[string]func(w http.ResponseWriter, r *http.Request){
+			"200_with_empty_data": func(w http.ResponseWriter, _ *http.Request) {
+				w.WriteHeader(http.StatusOK)
+			},
+			"400": func(w http.ResponseWriter, _ *http.Request) {
+				w.WriteHeader(http.StatusBadRequest)
+			},
+			"401": func(w http.ResponseWriter, _ *http.Request) {
+				w.WriteHeader(http.StatusUnauthorized)
+			},
+			"403": func(w http.ResponseWriter, _ *http.Request) {
+				w.WriteHeader(http.StatusForbidden)
+			},
+			"500": func(w http.ResponseWriter, _ *http.Request) {
+				w.WriteHeader(http.StatusInternalServerError)
+			},
+		}
+
+		for testCaseName, handlerFunc := range cases {
+			handlerFunc := handlerFunc
+			t.Run(testCaseName, func(t *testing.T) {
+				runHTTPTestServer(t, 3000, handlerFunc)
+
+				m := provider.ExternalMetricInfo{Metric: testMetricName}
+
+				if _, err := p.GetExternalMetric(ctx, "", nil, m); err == nil {
+					t.Errorf("Unexpected error = %v", err)
+				}
+
+				testutil.RetryGetRequestAndCheckStatus(ctx, t, testEnv.HTTPClient, url, func(statusCode int) bool {
+					return statusCode == http.StatusOK
+				})
+			})
+		}
+	})
+
+	t.Run("when_newrelic_backend_response_error_200_for_first_requests_and_then_500", func(t *testing.T) {
+		server := runHTTPTestServer(t, 3000, func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusOK)
+			_, _ = w.Write([]byte(`{"data": {
+				"actor": {
+				  "account": {
+					"nrql": {
+					  "results": [{"one": 0.015}]
+					}
+				  }
+				}
+			}}`))
+		})
+
+		m := provider.ExternalMetricInfo{Metric: testMetricName}
+
+		// Perform 2 requests (totally arbitrary number) returning 200 status code.
+		for i := 0; i < 2; i++ {
+			if _, err := p.GetExternalMetric(ctx, "", nil, m); err != nil {
+				t.Errorf("Unexpected error = %v", err)
+			}
+
+			testutil.RetryGetRequestAndCheckStatus(ctx, t, testEnv.HTTPClient, url, func(statusCode int) bool {
+				return statusCode != http.StatusOK
+			})
+		}
+
+		// Update handler to make the following requests to fail.
+		server.Config.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Content-Type", "application/json")
```

Comment:
why is this header here and not in the other tests? is this needed?

Reply:
No, forgot to remove it. Thanks